### PR TITLE
[INLONG-9138][Agent] Add task manager

### LINKED ISSUE #9138: [Improve][Agent] Add task manager
### Description

Add task manager

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
### Prepare a Pull Request
[INLONG-9138][Agent] Add task manager

- Fixes #9138 

### Motivation

Add task manager

### Modifications

Add task manager

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-core/src/test/java/org/apache/inlong/agent/core/AgentBaseTestsHelper.java [hunk 1]
```diff
@@ -64,4 +71,32 @@ public void teardownAgentHome() {
...
+            TaskStateEnum state) {
+        DataConfig dataConfig = getDataConfig(taskId, pattern, retry, startTime, endTime, state);
+        TaskProfile profile = TaskProfile.convertToTaskProfile(dataConfig);
+        return profile;
+    }
+
+    private DataConfig getDataConfig(int taskId, String pattern, boolean retry, Long startTime, Long endTime,
+            TaskStateEnum state) {
+        DataConfig dataConfig = new DataConfig();
+        dataConfig.setInlongGroupId("testGroupId"); // 老字段 groupId
```
Comment (doleyzi) [inline inlong-agent/agent-core/src/test/java/org/apache/inlong/agent/core/AgentBaseTestsHelper.java:85]: Chinese

**Review by doleyzi — APPROVED** [review]

File: inlong-agent/agent-core/src/test/java/org/apache/inlong/agent/core/AgentBaseTestsHelper.java (hunk 1 above)
Comment (justinwwhuang) [reply inlong-agent/agent-core/src/test/java/org/apache/inlong/agent/core/AgentBaseTestsHelper.java:85]: get

File: inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/task/file/TaskManager.java [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+        while (isRunnable()) {
+            try {
+                TaskAction action = queue.poll();
+                if (action == null) {
+                    break;
+                }
+                TaskProfile profile = action.getProfile();
+                switch (action.getActionType()) {
+                    case FINISH:
+                        LOGGER.info("test123 deal finish action, taskId {}", profile.getTaskId());
```
Comment (EMsnap) [inline inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/task/file/TaskManager.java:171]: what the meaning of test123

Comment (justinwwhuang) [reply inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/task/file/TaskManager.java:171]: for test, will delete it

**Review by EMsnap — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**